### PR TITLE
 Added camera settings test 

### PR DESCRIPTION
**Pull request by julianoes**
This is a first  skeleton, more testing will be added.

This runs against [camera-manager](https://github.com/Dronecode/camera-manager/) for now just using a local webcam.

### CODE REVIEW

**Comment by julianoes** [comment]
@bkueng I would appreciate a quick review, otherwise I'll just have to merge it.

File: src/main.cpp [hunk 1]
```diff
@@ -53,27 +46,32 @@ int main(int argc, char** argv)
...
+    System& system = dc.system();
+    system.register_component_discovered_callback([prom, &system](ComponentType component_type) {
+        std::cout << NORMAL_CONSOLE_TEXT << "Discovered a component with type "
+                  << unsigned(component_type) << std::endl;
+        try {
+            prom->set_value();
+            // We only need to receive this once, now we can unsubscribe again.
+            system.register_component_discovered_callback(nullptr);
+        } catch (const std::future_error& e) {
+            // Ignore if prom is set multiple times which can happen even though we're
```
Comment (bkueng) [inline src/main.cpp:61]: Is this still an issue if you unregister before triggering the promise?

File: src/tests/camera_settings.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    std::vector<Camera::Option> options;
+    _camera.get_possible_options(_config.param_name, options);
+
+    for (const auto& option : options) {
+        std::cout << "- " << option.option_description << " (" << option.option_id << ")"
+                  << std::endl;
+        setOption(_config.param_name, option);
+        Camera::Option set_option = getOption(_config.param_name);
+        EXPECT_EQ(set_option, option);
+        std::this_thread::sleep_for(std::chrono::seconds(1));
```
Comment (bkueng) [inline src/tests/camera_settings.cpp]: Why do we sleep here?

File: src/tests/camera_settings.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * Test setting and getting camera parameters.
+ */
+class CameraSettings : public TestBase
+{
+public:
+    struct Config {
+        unsigned camera_id{};
+        std::string param_name{};
+        std::string param_value{};
+        std::string sub_param_name{};
```
Comment (bkueng) [inline src/tests/camera_settings.h:21]: I don't see this being used?

**Review by bkueng — COMMENTED** [review]
Looks good, I have a remarked a couple of questions.

File: src/main.cpp (hunk 1 above)
Comment (julianoes) [reply src/main.cpp:61]: Unfortunately yes if the callback gets called in rapid succession.

File: src/tests/camera_settings.cpp (hunk 2 above)
Comment (julianoes) [reply src/tests/camera_settings.cpp]: I guess to see an effect when running the test but I suppose it's not necessary.

File: src/tests/camera_settings.h (hunk 3 above)
Comment (julianoes) [reply src/tests/camera_settings.h:21]: Not yet. I added a `FIXME`.

**Review by bkueng — APPROVED** [review]
Good to merge when CI passes.

**Pull request merged**